### PR TITLE
Allow engine to run without private keys

### PR DESCRIPTION
**Pull request by samholmes**
### CHANGELOG

- changed: Allow engine to run without private keys

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204112329611724

### CODE REVIEW

File: src/MoneroTools.js
Comment (swansontec) [inline src/MoneroTools.js:1]: Would the filename be lowercase `moneroTools.js`, to match what we do in accountbased?

File: yarn.lock [hunk 1]
```diff
@@ -6802,13 +6802,20 @@ yaml@^1.10.0:
...
   dependencies:
     rfc4648 "^1.1.0"
 
+yaob@^0.3.9:
+  version "0.3.10"
+  resolved "https://registry.yarnpkg.com/yaob/-/yaob-0.3.10.tgz#fb86d54d564ff6014f6f52fafc4219b6c068f613"
+  integrity sha512-QU+03xtNssGOw01Tten5zlfNcrsMPQjy0Nr/gueVEbXYsXHgYeDKXNLgPNMdYdTY7qkvLj8IGLOk/MKtsUh+Fg==
+  dependencies:
+    rfc4648 "^1.1.0"
+
```
Comment (swansontec) [inline yarn.lock:6818]: We should delete yaob from package.json, since we don't actually use it anywhere in this repo.

**Review by swansontec — APPROVED** [review]
Approved, but we should add another commit at the very beginning, "Remove unused yaob dependency from package.json"

File: src/MoneroTools.js
Comment (samholmes) [reply src/MoneroTools.js:1]: The class name is `MoneroTools`, not `moneroTools`. If this file wasn't exporting a class (type) like `export class MoneroTools` or `export const makeMoneroTools`, then I'd say yes.

Comment (samholmes) [reply src/MoneroTools.js:1]: Cool. To expand a bit more my thought process when I read file names: 

If I see a file name that is camelCase, then I know it's a file that may export a bunch of things related to the name of the file. E.g. `utils/serverUtil.ts` or `db/users.ts`. 

If I see a PascalCase file name, then I know it exports primarily a class or type of some kind. E.g. `components/ButtonModal` or `utils/PeriodicTask.ts`. The technique used by the type whether it be a class or factory function (`makeThing`) is not relevant to knowing that the file exports this type which can have many instances.

Another way of thinking about it perhaps is that camelCase are for singleton exports, whereas PascalCase are for multitons. But, I haven't evaluated all the cases to say that this is for sure the pattern here, so don't quote me on this.

**Review by swansontec — APPROVED** [review]
We

**Pull request merged**